src/nwp_consumer/internal/inputs/noaa/ncar.py

```diff
@@ -143,50 +113,55 @@ def mapCachedRaw(self, *, p: pathlib.Path) -> xr.Dataset:  # noqa: D102
+            if any(x in d.coords for x in ["surface", "heightAboveGround", "isobaricInhPa"])
         ]
 
         # Split into surface, heightAboveGround, and isobaricInhPa lists
-        surface = [d for d in ds if "surface" in d.coords]
-        heightAboveGround = [d for d in ds if "heightAboveGround" in d.coords]
-        isobaricInhPa = [d for d in ds if "isobaricInhPa" in d.coords]
+        surface: list[xr.Dataset] = [d for d in ds if "surface" in d.coords]
+        heightAboveGround: list[xr.Dataset] = [d for d in ds if "heightAboveGround" in d.coords]
+        isobaricInhPa: list[xr.Dataset] = [d for d in ds if "isobaricInhPa" in d.coords]
+        del ds
 
         # Update name of each data variable based off the attribute GRIB_stepType
         for i, d in enumerate(surface):
-            for variable in d.data_vars.keys():
+            for variable in d.data_vars:
                 d = d.rename({variable: f"{variable}_surface_{d[f'{variable}'].attrs['GRIB_stepType']}"})
             surface[i] = d
         for i, d in enumerate(heightAboveGround):
-            for variable in d.data_vars.keys():
+            for variable in d.data_vars:
                 d = d.rename({variable: f"{variable}_{d[f'{variable}'].attrs['GRIB_stepType']}"})
             heightAboveGround[i] = d
 
-        surface = xr.merge(surface)
-        # Drop unknown data variable
-        surface = surface.drop_vars("unknown_surface_instant", errors="ignore")
-        heightAboveGround = xr.merge(heightAboveGround)
-        isobaricInhPa = xr.merge(isobaricInhPa)
+        surface_merged: xr.Dataset = xr.merge(surface).drop_vars(
+            ["unknown_surface_instant", "valid_time"], errors="ignore",
+        )
+        del surface
+        heightAboveGround_merged: xr.Dataset = xr.merge(heightAboveGround).drop_vars(
+            ["valid_time"], errors="ignore",
+        )
+        del heightAboveGround
+        isobaricInhPa_merged: xr.Dataset = xr.merge(isobaricInhPa).drop_vars(
+            ["valid_time"], errors="ignore",
+        )
+        del isobaricInhPa
 
-        ds = xr.merge([surface, heightAboveGround, isobaricInhPa])
+        total_ds = xr.merge([surface_merged, heightAboveGround_merged, isobaricInhPa_merged])
+        del surface_merged, heightAboveGround_merged, isobaricInhPa_merged
 
         # Map the data to the internal dataset representation
         # * Transpose the Dataset so that the dimensions are correctly ordered
         # * Rechunk the data to a more optimal size
-        ds = (
-            ds.rename({"time": "init_time"})
+        total_ds = (
+            total_ds.rename({"time": "init_time"})
             .expand_dims("init_time")
             .expand_dims("step")
             .transpose("init_time", "step", ...)
             .sortby("step")
-            .chunk(
-                {
-                    "init_time": 1,
-                    "step": -1,
-                },
-            )
+            .chunk({"init_time": 1, "step": 1})
```

Comment:
do you want to chuck it a bit more with step?

Reply:
I can't - there's only one step per file and they won't merge if I step differently!